### PR TITLE
Add joinPayload and leave channel when channel is changed

### PR DESCRIPTION
**Pull request by indocomsoft**

### CODE REVIEW

File: Web/src/components/room/hooks/useErrorableChannel.ts [hunk 1]
```diff
@@ -41,13 +43,19 @@ export default function useErrorableChannel(channelId: string): ErrorableChannel
...
+          setChannel(oldChannel => {
+            if (oldChannel != null) oldChannel.leave();
+            return newChannel;
+          });
+        })
         .receive(ChannelResponse.ERROR, reasons => setError([ErrorCause.Other, reasons]))
         .receive(ChannelResponse.TIMEOUT, () => setError([ErrorCause.Timeout, null]));
 
       return currentSocket;
     });
```
Comment (juxd) [inline Web/src/components/room/hooks/useErrorableChannel.ts]: is there some channel cleanup stuff here we need to do when we stop using this channel? e.g. `channel.disconnect()`

Comment (indocomsoft) [reply Web/src/components/room/hooks/useErrorableChannel.ts]: Yes, it's on line 49: `oldChannel.leave()`

Comment (indocomsoft) [reply Web/src/components/room/hooks/useErrorableChannel.ts]: Yes, it's on line 49: `oldChannel.leave()`

Comment (juxd) [reply Web/src/components/room/hooks/useErrorableChannel.ts]: Ah, ok.

**Pull request merged**